### PR TITLE
Refactored test case plugin loading and unloading.

### PR DESCRIPTION
**Pull request by ppt-adsk**
There was duplication of infrastructure for loading plugins needed for a test, and unloading them afterwards.  This has been removed, and the union of the support was kept, including the `_requiredPlugins` Python class variable that is presented in the Markdown documentation.

### CODE REVIEW

File: test/lib/mayaUsd/render/mayaToHydra/cpp/testDataProducerExample.py [hunk 1]
```diff
@@ -21,32 +21,14 @@ class TestDataProducerExample(mtohUtils.MayaHydraBaseTestCase):
...
-    _pluginsToUnload = []
+    _requiredPlugins = ['mayaHydraCppTests', 'mayaHydraFlowViewportAPILocator']
 
     def createScene(self):
         self._locator = cmds.createNode('MhFlowViewportAPILocator')
         cmds.setAttr(self._locator + '.numCubesX', 3)
         cmds.setAttr(self._locator + '.numCubesY', 3)
         cmds.setAttr(self._locator + '.numCubesZ', 3)
 
-    @classmethod
```
Comment (ppt-adsk) [inline test/lib/mayaUsd/render/mayaToHydra/cpp/testDataProducerExample.py:33]: This is the duplication, brought back to the base class.

File: test/testUtils/mtohUtils.py [hunk 2]
```diff
@@ -94,6 +97,12 @@ def setUp(self):
 
     @classmethod
     def tearDownClass(cls):
```
Comment (ppt-adsk) [inline test/testUtils/mtohUtils.py:104]: Base class now unloads plugins that were loaded specifically for the test.

File: test/testUtils/mtohUtils.py [hunk 3]
```diff
@@ -94,6 +97,12 @@ def setUp(self):
...
+        # Clean out the scene to allow all plugins to unload cleanly.
+        cmds.file(new=True, force=True)
+        for p in reversed(cls._pluginsToUnload):
+            if p != 'mayaHydraFlowViewportAPILocator':
```
Comment (ppt-adsk) [inline test/testUtils/mtohUtils.py]: Unloading this plugin crashes, entered as HYDRA-1304.

**Review by lilike-adsk — DISMISSED** [review]
Nice cleanup!

**Review by lilike-adsk — APPROVED** [review]

**Pull request merged**